python_modules/dagster/dagster/core/definitions/graph_definition.py

```diff
@@ -646,6 +650,15 @@ def __init__(
             tags=parent_graph_def.tags,
         )
 
+    @property
+    def parent_graph_def(self) -> GraphDefinition:
+        return self._parent_graph_def
+
+    def get_top_level_ignored_nodes(self) -> List[Node]:
+        return [
+            solid for solid in self.parent_graph_def.solids if not self.has_solid_named(solid.name)
+        ]
```

Comment:
I think `original` might be a better name than `parent` - debatable 

the "ignored" language feels off to me from the perspective of this object - maybe `get_top_level_omitted_nodes` or something like that? 

definitely deserves a doc block

Reply:
Re: original vs parent, I did switch it back and forth. 

I ended up naming it `parent` because it currently refers to the direct parent rather than the root, while `original` may sound more like a "root".

-----

naming aside, why parent not root?
It logs the direct parent because the graph difference (i.e. the nodes that `get_top_level_ignored_nodes` returns) is between an operating graph and its direct parent.

here's an illustration of the current implementation:
![image](https://user-images.githubusercontent.com/4531914/149036052-f639ab11-8342-492c-b202-445b15c935f4.png)

I can see we record the root/original graph def instead, then the pic would be:
![image](https://user-images.githubusercontent.com/4531914/149036514-9d8f23b4-bde7-4e35-b7cb-53ada6032347.png)

but we will need to piggyback the user-input selection of Job 2 to Job 3 so we can build the new graph on top of subselected graph. This wasn't a question bc we never allowed subselect a subset. I think how we thread that info would depend on how we'd resolve the query (which i haven't thought through yet) so i'm keeping this lineage simple to just log the direct parent for now (and i think for the config validation skipping nodes, parent vs root doesn't matter) - open to suggestions.

https://excalidraw.com/#json=3WjGMKlTSXx3KVVBj08EI,YGBQLRGK6tOF5b2SqC7Jfw for reference

Reply:
Ah i see, probably worth leaving a comment that captures this idea somewhere

Reply:
still think `omitted` (or something like that) is better than `ignored` tho